src/quotamodel.c

```diff
@@ -900,6 +897,75 @@ merge_uncommitted_table_to_oidlist(List *oidlist)
 	return oidlist;
 }
 
+static TableSizeEntry *
+get_tsentry(Oid oid, int16 segid)
+{
+	bool              found;
+	TableSizeEntryKey key     = {.reloid = oid, .id = TableSizeEntryId(segid)};
+	HASHACTION        action  = check_hash_fullness(table_size_map, MAX_NUM_TABLE_SIZE_ENTRIES, table_size_map_warning,
+	                                                table_size_map_last_overflow_report);
+	TableSizeEntry   *tsentry = hash_search(table_size_map, &key, action, &found);
+
+	if (!found && tsentry != NULL)
+	{
+		memset(tsentry->totalsize, 0, sizeof(tsentry->totalsize));
+		tsentry->owneroid      = InvalidOid;
+		tsentry->namespaceoid  = InvalidOid;
+		tsentry->tablespaceoid = InvalidOid;
+		tsentry->flag          = 0;
+
+		int seg_st = TableSizeEntrySegidStart(tsentry);
+		int seg_ed = TableSizeEntrySegidEnd(tsentry);
+		for (int j = seg_st; j < seg_ed; j++) TableSizeEntrySetFlushFlag(tsentry, j);
+	}
+
+	/* mark tsentry is_exist */
+	if (tsentry) set_table_size_entry_flag(tsentry, TABLE_EXIST);
+
+	return tsentry;
+}
+
+void
+update_active_table_size(Oid oid, int64 size, int16 segid)
+{
+	TableSizeEntry *tsentry = get_tsentry(oid, segid);
+
+	if (tsentry == NULL)
+	{
+		/* Too many tables have been added to the table_size_map, to avoid diskquota using
+		   too much share memory, just return. The diskquota won't work correctly
+		   anymore. */
```

Comment:
According to the comment 'The diskquota won't work correctly anymore.'  it looks like an error. May be raise an ERROR here instead of silent return?

Reply:
This has nothing to do with the current patch.

Reply:
This was also the case before the patch.

Reply:
This comment now appeared twice. The case was only one.